py/desispec/spectra.py

```diff
@@ -128,9 +139,12 @@ def __init__(self, bands=[], wave={}, flux={}, ivar={}, mask=None,
                 if resolution_data[b].shape[2] != wave[b].shape[0]:
                     raise RuntimeError("resolution array wavelength dimension for band {} does not match grid".format(b))
             if extra is not None:
-                for ex in extra[b].items():
-                    if ex[1].shape != flux[b].shape:
-                        raise RuntimeError("extra arrays must have the same shape as the flux array")
+                try:
+                    for ex in extra[b].items():
+                        if ex[1].shape != flux[b].shape:
+                            raise RuntimeError("extra arrays must have the same shape as the flux array")
+                except (KeyError, AttributeError):
+                    pass
```

Comment:
Please explain the case where this generates KeyError or AttributeError but is otherwise ok and should be silently passed.

Reply:
The `except` is catching errors thrown by line 143 (which is not in the snippet above).

* If `extra` exists, but `extra[b]` does not exist, line 143 will throw `KeyError`.
* If `extra[b]` exists, but is `None`, line 143 will throw `AttributeError`.